### PR TITLE
exome-seq - GATK4 Preprocessing single-sample workflow

### PR DESCRIPTION
**Pull request by dleehr**
Adds GATK4 tools and a preprocessing workflow for exome-seq based on tools and parameters in WDL/JSON from GATK best practices at https://github.com/gatk-workflows/gatk4-data-processing

### CODE REVIEW

**Comment by dleehr** [comment]
Also includes some tweaks to exome-seq and a workflow to wrap just the single-sample processing part of the GATK3 workflow

File: tools/gotc-bwa-mem-samtools.cwl [hunk 1]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+
+requirements:
+- class: DockerRequirement
+  dockerPull: 'broadinstitute/genomes-in-the-cloud:2.3.1-1512499786'
```
Comment (johnbradley) [inline tools/gotc-bwa-mem-samtools.cwl]: Filename is `gotc-...` but the image is `genomes-in-the-cloud`?

File: subworkflows/exomeseq-gatk4-01-preprocessing.cwl [hunk 2]
```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: Workflow
+requirements:
+  - class: ScatterFeatureRequirement
+  - $import: ../types/bespin-types.yml
+inputs:
+  # NOTE: How long is this expected to take?
```
Comment (johnbradley) [inline subworkflows/exomeseq-gatk4-01-preprocessing.cwl]: There are several question comments like this one in this file.

File: tools/gotc-bwa-mem-samtools.cwl (hunk 1 above)
Comment (dleehr) [reply tools/gotc-bwa-mem-samtools.cwl]: Yep. ¯\\_(ツ)_/¯

File: subworkflows/exomeseq-gatk4-01-preprocessing.cwl (hunk 2 above)
Comment (dleehr) [reply subworkflows/exomeseq-gatk4-01-preprocessing.cwl]: Will clean those up. Thanks!

File: tools/gotc-bwa-mem-samtools.cwl (hunk 1 above)
Comment (dleehr) [reply tools/gotc-bwa-mem-samtools.cwl]: Sorry for the ambiguous reply. GATK refers to this as gotc in their [WDL](https://github.com/gatk-workflows/gatk4-data-processing/blob/8ffa26ff4580df4ac3a5aa9e272a4ff6bab44ba2/processing-for-variant-discovery-gatk4.wdl) and corresponding [JSON](https://github.com/gatk-workflows/gatk4-data-processing/blob/8ffa26ff4580df4ac3a5aa9e272a4ff6bab44ba2/processing-for-variant-discovery-gatk4.b37.wgs.inputs.json).

I thought I was carrying that forward for consistency, but I realize that `gotc` doesn't appear anywhere else in my CWL repo, so there's no good reason to use it.

**Pull request merged**